scrapy/core/engine.py

```diff
@@ -5,47 +5,57 @@
 
 """
 import logging
+import warnings
 from time import time
+from typing import Callable, Iterable, Iterator, Optional, Set, Union
 
-from twisted.internet import defer, task
+from twisted.internet.defer import Deferred, inlineCallbacks, succeed as defer_succeed
```

Comment:
Note that we have defer_succeed in utils.defer

Reply:
`utils.defer.defer_succeed` introduces a bit of delay, this renaming is because the previous usage was `defer.succed`, and I thought just `succeed` might not be as intuitive. I'll switch back to `succeed` to avoid confusion, thanks.

Reply:
Sure, I just meant that there can be confusion and also a theoretical naming clash in the future.